Services/CO.CDP.DataSharing.WebApi/Api/DataSharing.cs

```diff
@@ -174,24 +176,18 @@ public static void UseDataSharingEndpoints(this WebApplication app)
                 return operation;
             });
 
-        app.MapPost("/share/data", (ShareRequest shareRequest) => Results.Ok(new ShareReceipt
-        {
-            ShareCode = Guid.NewGuid().ToString(),
-            ExpiresAt = shareRequest.ExpiresAt,
-            FormId = shareRequest.SupplierFormId,
-            FormVersionId = "20240317",
-            Permissions = shareRequest.Permissions
-        }
-            ))
+        app.MapPost("/share/data", async (ShareRequest shareRequest, IUseCase<ShareRequest, ShareReceipt?> useCase) =>
```

Comment:
Should be `IUseCase<ShareRequest, ShareReceipt>`

Reply:
done.